src/app/components/utility/modal/modal.component.ts

```diff
@@ -255,6 +266,9 @@ export class ModalComponent implements OnInit {
         this.inputErrorMessage = 'Password do not match';
       }
     }
+    if (this.user.google_scholar_url !== this.urlval || this.user.github_url !== this.urlval || this.user.linkedin_url !== this.urlval) {
```

Comment:
This doesn't seems good, please take a look how to use regex.

Reply:
Where should I put it?

Reply:
I can't understand

Reply:
You can take a look to this PR (https://github.com/Cloud-CV/EvalAI-ngx/pull/231). For consistency purpose we should use the same regex.

Reply:
i have implemented the regex code that was suggested in that pr but now if i write a url with correct syntax it doesn't remove please enter a valid url.

Reply:
i have pushed my code

Reply:
You also need to call `regex.test`. Like for checking `googleScholarUrl` you need to call `regex.test(googleScholarUrl)`. If it is a valid url then it will return `true`

Reply:
Also please use this regex `^(http[s]?:\\/\\/(www\\.)?|ftp:\\/\\/(www\\.)?|www\\.){1}([0-9A-Za-z-\\.@:%_\+~#=]+)+((\\.[a-zA-Z]{2,3})+)(/(.)*)?(\\?(.)*)?`

Reply:
![evalai-profileerror](https://user-images.githubusercontent.com/34577232/70844731-fa202a00-1e6a-11ea-8ba0-40f680146104.png)

Reply:
it is not detecting them

Reply:
Cause you have changed `google_scholar_url` to `googleScholarUrl` :smile:

Reply:
to change this i also have to change the name variable in const params?

Reply:
this code 
 const firstName = (this.user['first_name'] === '-') ? '' : this.user['first_name'];
it affecting all the variables in backend and frontend?

Reply:
so i also have to change the front end html code right?

Reply:
no it's not. It's just a temporary variable for holding the `firstName` value and then assigning it to the form variable which is `value`.

Reply:
> so i also have to change the front end html code right?

no, you don't have to do that.

Reply:
Also use `const` as declaration for regex.

Reply:
sir it is still the same . i have pushed my code so you can look in it

Reply:
Instead of checking on the input event change, how about checking url validation after clicking submit button in `updateUserDetails` function?

Reply:
so

 if (confirm == submit ){
if(regex.test(googleScholarUrl) !== true  ){
show error message;
}

Reply:
how will i get the value of googleScholarUrl ??

Reply:
if i will pass in just googleScholarUrl then the const which we declared above will be called?

Reply:
and when the submit button is pressed the form closes so there is no use of url validation in updateuserdetails

Reply:
When the submit pressed then this (https://github.com/Cloud-CV/EvalAI-ngx/blob/master/src/app/components/utility/modal/modal.component.ts#L192) `formValidate` function gets called. Please take a look how we came to know that this call is for `Update profile` and then how can we add validation and stop from closing the modal .

Reply:
sir i am completing now before going to rest. but i am facing a problem. i have put a if condition in formvalidater . but it requires a variable to be called from profile.componet so . how can i perform that

Reply:
Which variable is required?

Reply:
like i have created a urlval var to store a true or false and i need it in model.compnent

Reply:
and also what this.confirmed(this) do?

Reply:
> and also what this.confirmed(this) do?

When validation is done then `confirmed` function gets called.

Reply:
and what should i write to stop this because in the if conditions of formvalidator there is only a console.log function . so how you stop the form from being submitted?

Reply:
You just don't have to do anything, I mean just don't call the `this.confirmed`

Reply:
ok sir i got it but how i call the variable from profile.component to model.component . i tried the export method with import method but that did'nt work

Reply:
but why you want to call some variable from `profile.component`?

Reply:
Url is checked in profile.component and a true or false is assigned to a variable name urlval and it is needed for if condition in model.component . As we have to keep the url Verification in profile.component only

Reply:
You can do the url validation in `modal.component` also.

Reply:
sir i tried making the submit button stop when url is not valide but it is not working . i am pushing my code so you can take look

Reply:
i found the problem sir the user.googlescholarurl is giving the test an undefined value

Reply:
sir the user variable in the modal.component is containing the values that are not changing and because of this my if statement is getting a '-' hyphen in return instead of a url

Reply:
Yeah cause it is the stored value not the one which is on the input.

Why don't you check all the url field validation here (https://github.com/Cloud-CV/EvalAI-ngx/pull/234/files#diff-102b6aae81d792250ba74b4d834949d3R195)

Reply:
how can i check for the value of url when the pre build user variable is not updating

Reply:
Please console `this.formComponents`, you can get the current input value from there.

Reply:
it is giving the input values but . it is giving them in 6 different type of input components and how can i access them. i tried this console.log(this.formComponents._results) which give 6 input component . but i am unable to access the specific value of urls fields.

Reply:
Yes, it is an array. Just access it by index and get it's `value`. Like for `first_name` > `this.formComponents._results[0].value`